bigquery/api/GettingStarted/BigqueryTest/BigqueryTest.cs

```diff
@@ -533,18 +537,36 @@ public void TestImportDataFromFile()
         public void TestImportDataFromStream()
         {
             string datasetId = "datasetForTestImportDataFromStream";
-            string newTableId = "tableForTestImportDataFromStream";
+            Random rnd = new Random();
+            string tableNameSuffix = rnd.Next(4200000).ToString();
+            string newTableId = "tableForTestImportDataFromStream" + tableNameSuffix;
             string gcsUploadTestWord = "exampleJsonFromStream";
+            string valueToTest = "";
             CreateDataset(datasetId, _client);
             CreateTable(datasetId, newTableId, _client);
             // Import data.
-            UploadJson(datasetId, newTableId, _client);
+            UploadJsonStreaming(datasetId, newTableId, _client);
             // Query table to get first row and confirm it contains the expected value.
             var newTable = _client.GetTable(datasetId, newTableId);
             string query = $"SELECT title, unique_words FROM {newTable} ORDER BY title";
-            BigQueryResults results = AsyncQuery(_projectId, datasetId, newTableId, query, _client);
-            var row = results.First();
-            Assert.Equal(gcsUploadTestWord, row["title"]);
+            var retryPolicy = new
+                RetryPolicy<CustomTransientErrorDetectionStrategy>
+                (RetryStrategy.DefaultExponential);
+            try
+            {
+                retryPolicy.ExecuteAction(() =>
+                {
+                    BigQueryResults results = AsyncQuery(_projectId, datasetId,
+                        newTableId, query, _client);
+                    var row = results.First();
+                    valueToTest = row["title"].ToString();
+                });
+            }
+            catch (Exception)
+            {
+                // All of the retries failed.
+            }
+            Assert.Equal(gcsUploadTestWord, valueToTest);

```

Comment:
If this assert fails, then DeleteTable and DeleteDataset won't be called below.  The test will leak resources.